python/postprocessing/modules/common/muonScaleResProducer.py

```diff
@@ -0,0 +1,60 @@
+import ROOT
+import os
+import random
+ROOT.PyConfig.IgnoreCommandLineOptions = True
+
+from PhysicsTools.NanoAODTools.postprocessing.framework.datamodel import Collection
+from PhysicsTools.NanoAODTools.postprocessing.framework.eventloop import Module
+
+class muonScaleResProducer(Module):
+    def __init__(self, rc_dir, rc_corrections):
+        p_postproc = '%s/src/PhysicsTools/NanoAODTools/python/postprocessing' % os.environ['CMSSW_BASE']
+        p_roccor = p_postproc + '/data/' + rc_dir
+        if "/RoccoR_cc.so" not in ROOT.gSystem.GetLibraries():
+            p_helper = '%s/RoccoR.cc' % p_roccor
+            print 'Loading C++ helper from ' + p_helper
+            ROOT.gROOT.ProcessLine('.L ' + p_helper)
+        self._roccor = ROOT.RoccoR(p_roccor + '/' + rc_corrections)
+
+    def beginJob(self):
+        pass
+
+    def endJob(self):
+        pass
+
+    def beginFile(self, inputFile, outputFile, inputTree, wrappedOutputTree):
+        self.out = wrappedOutputTree
+        self.out.branch("Muon_pt_corrected", "F", lenVar="nMuon")
+        # self.out.branch("Muon_pt_err", "F", lenVar="nMuon")
+
+    def endFile(self, inputFile, outputFile, inputTree, wrappedOutputTree):
+        pass
+
+    def analyze(self, event):
+        muons = Collection(event, "Muon")
+        roccor = self._roccor
+        try:
+            is_mc = bool(getattr(event, 'GenJet_pt'))
+        except RuntimeError:
+            is_mc = False
```

Comment:
can the is_mc be derived once per job in beginFile, checking if inputTree.GetBranch("GenJet_pt") is there?

Reply:
Thanks!
(just pushed.. )